jaeger-core/src/main/java/com/uber/jaeger/Tracer.java

```diff
@@ -438,25 +442,21 @@ public Scope startActive(boolean finishSpanOnClose) {
    * Builds Jaeger Tracer with options.
    */
   public static final class Builder {
-    private final Sampler sampler;
-    private final Reporter reporter;
+    private Sampler sampler;
+    private Reporter reporter;
     private final PropagationRegistry registry = new PropagationRegistry();
     private Metrics metrics = new Metrics(new StatsFactoryImpl(new NullStatsReporter()));
+    // not final - make it overridable
     private String serviceName;
     private Clock clock = new SystemClock();
     private Map<String, Object> tags = new HashMap<String, Object>();
     private boolean zipkinSharedRpcSpan;
     private ScopeManager scopeManager = new ThreadLocalScopeManager();
     private BaggageRestrictionManager baggageRestrictionManager = new DefaultBaggageRestrictionManager();
+    private boolean withLoggingSpans;
 
-    public Builder(String serviceName, Reporter reporter, Sampler sampler) {
-      if (serviceName == null || serviceName.trim().length() == 0) {
-        throw new IllegalArgumentException("serviceName must not be null or empty");
-      }
-      this.serviceName = serviceName;
-      this.reporter = reporter;
-      this.sampler = sampler;
-
+    public Builder(String serviceName) {
+      this.withServiceName(serviceName);
```

Comment:
Q: what is the scenario where withServiceName() method is needed instead of providing the name to the ctor?